pkgs/build-support/substitute/substitute-all.nix

```diff
@@ -1,12 +1,20 @@
-{ stdenvNoCC }:
-
-args:
-
+{ lib, stdenvNoCC }:
 # see the substituteAll in the nixpkgs documentation for usage and constraints
-stdenvNoCC.mkDerivation ({
-  name = if args ? name then args.name else baseNameOf (toString args.src);
-  builder = ./substitute-all.sh;
-  inherit (args) src;
-  preferLocalBuild = true;
-  allowSubstitutes = false;
-} // args)
+args:
+let
+  # keep this in sync with substituteAll
+  isInvalidArgName = x: builtins.match "^[a-z][a-zA-Z0-9_]*$" x == null;
```

Comment:
At least in bash variable names can also start with underscores and can be one character long.

Reply:
That's the exact regex `substituteAll` uses.

Reply:
Then we shouldn't write below that it must be a valid shell variable name.

Reply:
Yeah, reworded the message to actually describe the requirements.